### ISSUE TITLE
Error in accurev GUI client - JAB

### ISSUE THREAD

**Issue by manish10** [nvaccess/nvda#5614]
NVDA used to work well with the accurev GUI client till the previous release. After release 2015.4, NVDA fails to speak most of the items in accurev when tabbing around or through object navigation. The nvda.log file has the following error repeated over and over:

```
ERROR - eventHandler.executeEvent (09:58:52):
error executing event: gainFocus on <NVDAObjects.Dynamic_EditableTextWithoutAutoSelectDetectionJAB object at 0x06385270> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyo", line 143, in executeEvent
  File "eventHandler.pyo", line 91, in __init__
  File "eventHandler.pyo", line 98, in next
  File "NVDAObjects\JAB\__init__.pyo", line 506, in event_gainFocus
  File "NVDAObjects\__init__.pyo", line 860, in event_gainFocus
  File "NVDAObjects\JAB\__init__.pyo", line 454, in reportFocus
  File "NVDAObjects\__init__.pyo", line 804, in reportFocus
  File "speech.pyo", line 357, in speakObject
  File "speech.pyo", line 274, in speakObjectProperties
  File "baseObject.pyo", line 34, in __get__
  File "baseObject.pyo", line 110, in _getPropertyViaCache
  File "NVDAObjects\JAB\__init__.pyo", line 240, in _get_keyboardShortcut
IndexError: invalid index
ERROR - eventHandler.executeEvent (09:58:54):
error executing event: gainFocus on <NVDAObjects.JAB.JAB object at 0x0641FBD0> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyo", line 143, in executeEvent
  File "eventHandler.pyo", line 91, in __init__
  File "eventHandler.pyo", line 98, in next
  File "NVDAObjects\JAB\__init__.pyo", line 506, in event_gainFocus
  File "NVDAObjects\__init__.pyo", line 860, in event_gainFocus
  File "NVDAObjects\JAB\__init__.pyo", line 454, in reportFocus
  File "NVDAObjects\__init__.pyo", line 804, in reportFocus
  File "speech.pyo", line 357, in speakObject
  File "speech.pyo", line 274, in speakObjectProperties
  File "baseObject.pyo", line 34, in __get__
  File "baseObject.pyo", line 110, in _getPropertyViaCache
  File "NVDAObjects\JAB\__init__.pyo", line 240, in _get_keyboardShortcut
IndexError: invalid index
```

**Comment by jcsteh** [comment]
There hasn't been any change to the JAB code since 2014. Are you absolutely certain that something else didn't change on your system; e.g. setting or JRE/JAB version? Have you re-tested with NVDA 2015.3 since you discovered the problem?

Of course, the fact that there was no JAB code change doesn't mean it isn't a bug introduced in the 2015.4 code. It does unfortunately mean I have no idea where to start unless I can reproduce this.

**Comment by manish10** [comment]
This is my office machine. It is very possible that a new version of something was pushed to this machine without my knowing it.
The accurev version and the JRE it uses hasn't changed.
I will try with NVDA 2015.3 tomorrow and update the ticket.

-Manish

-----Original Message-----

**Comment by manish10** [comment]
I tested with nvda 2015.2 and the problem is the same. So, it is something in my environment and not in the NVDA code. They probably deployed some new accruve version without letting me know.

You can close the bug as invalid.

-Manish

-----Original Message-----
